maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -3114,9 +3114,8 @@ protected void warnIfIllegalFailOnFlakeCount() throws MojoFailureException
 
     private void printDefaultSeedIfNecessary()
     {
-        if ( getRunOrderRandomSeed() == null && getRunOrder().equals( RunOrder.RANDOM.name() ) )
+        if ( getRunOrder().equals( RunOrder.RANDOM.name() ) )
         {
-            setRunOrderRandomSeed( System.nanoTime() );
```

Comment:
I think you wanted to have the following:

```
if ( getRunOrderRandomSeed() == null )
{
    setRunOrderRandomSeed( System.nanoTime() );
}
```

Reply:
Done